client/packages/requisitions/src/ResponseRequisition/api/hooks/document/useDeleteSelectedResponseRequisitions.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  useTranslation,
+  useTableStore,
+  RequisitionNodeStatus,
+  useDeleteConfirmation,
+  useUrlQueryParams,
+  useNotification,
+} from '@openmsupply-client/common';
+import { ResponseFragment } from '../../operations.generated';
+import { useDeleteResponses } from './useDeleteResponses';
+import { useResponses } from './useResponses';
+
+export const useDeleteSelectedResponseRequisitions = () => {
+  const { queryParams } = useUrlQueryParams({
+    initialSort: { key: 'createdDatetime', dir: 'desc' },
+  });
+  const { data: rows } = useResponses(queryParams);
+  const { mutateAsync } = useDeleteResponses();
+  const { info } = useNotification();
+  const t = useTranslation();
+  const { selectedRows } = useTableStore(state => ({
+    selectedRows: Object.keys(state.rowState)
+      .filter(id => state.rowState[id]?.isSelected)
+      .map(selectedId => rows?.nodes?.find(({ id }) => selectedId === id))
+      .filter(Boolean) as ResponseFragment[],
+  }));
+  const deleteAction = async () => {
+    let result = await mutateAsync(selectedRows).catch(err => {
+      throw err;
+    });
+    let errorMessages = [];
+    // check for errors
+    result.forEach(line => {
+      if (line.response.__typename == 'DeleteResponseRequisitionError') {
+        info(line.response.error.description)();
+        errorMessages.push(line.response.error.description);
+      }
+    });
+    if (errorMessages.length > 0) {
+      throw new Error();
+    }
```

Comment:
I've added a check here for errors on delete action to throw errors for each failed delete with structure rather than one error with the generic hook.

However it does leave an extra snack when error handling. But figure that is OK rather than more extended refactoring

Reply:
I'm getting double toast on error
![Screenshot 2024-11-06 at 9 09 06 AM](https://github.com/user-attachments/assets/9640c4ac-0776-4f2c-8bf0-f2df00ccdbd2)